### PR TITLE
gradle: Ensure git command is run projectDir rather than CWD.

### PR DESCRIPTION
**Pull request by vanitasvitae**
This commit is the same change as jonathan-haubrich authored for Smack
here: https://github.com/igniterealtime/Smack/commit/7a5886279473c06d361d2903ee60f8b0eb0d13f9

### CODE REVIEW

**Comment by Flowdalic** [comment]
I think this is a good candidate for jXMPP's stable branch. Therefore: Could you change the target branch of this PR to '1.0'?

File: build.gradle [hunk 1]
```diff
@@ -387,8 +387,9 @@ def getGitCommit() {
 	def dotGit = new File("$projectDir/.git")
 	if (!dotGit.isDirectory()) return 'non-git build'
 
+	def projectDir = dotGit.getParentFile()
```
Comment (Flowdalic) [inline build.gradle:390]: I find that kind of strange, after all, we just used $projectDir a few lines above. I wonder if there is a reason why the original author did this. Maybe

```
def proc = cmd.execute(null, "$projectDir")
```

does not work?

Comment (Flowdalic) [reply build.gradle:390]: Please change this to  follow https://github.com/igniterealtime/Smack/commit/45e865757a7242ea4942855213b53a9670c06f9c

**Comment by Flowdalic** [comment]
I think this is a good candidate for jXMPP's stable branch. Therefore: Could you change the target branch of this PR to '1.0'?

File: build.gradle (hunk 1 above)
Comment (vanitasvitae) [reply build.gradle:390]: Sure

Comment (Flowdalic) [reply build.gradle]: I still don't see a

```
def projectDirFile = new File("$projectDir")
```

Comment (vanitasvitae) [reply build.gradle]: Ah that's what you meant. I thought you were referring to the variable name alone.

**Pull request merged**